client/network/src/light_client_requests/handler.rs

```diff
@@ -0,0 +1,382 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Helper for incoming light client requests.
+//!
+//! Handle (i.e. answer) incoming light client requests from a remote peer received via
+//! [`crate::request_responses::RequestResponsesBehaviour`] with [`LightClientRequestHandler`].
+
+use codec::{self, Encode, Decode};
+use crate::{
+	chain::Client,
+	config::ProtocolId,
+	schema,
+	PeerId,
+};
+use crate::request_responses::{IncomingRequest, ProtocolConfig};
+use futures::{channel::{mpsc, oneshot},  prelude::*};
+use prost::Message;
+use sc_client_api::{
+	StorageProof,
+	light
+};
+// use sc_peerset::{PeersetHandle, ReputationChange};
+use sp_core::{
+	storage::{ChildInfo, ChildType,StorageKey, PrefixedStorageKey},
+	hexdisplay::HexDisplay,
+};
+use sp_runtime::{
+	traits::{Block, Zero},
+	generic::BlockId,
+};
+use std::{
+	collections::{BTreeMap},
+	sync::Arc,
+};
+use log::debug;
+
+const LOG_TARGET: &str = "light-client-request-handler";
+
+/// Handler for incoming light client requests from a remote peer.
+pub struct LightClientRequestHandler<B: Block> {
+	request_receiver: mpsc::Receiver<IncomingRequest>,
+	/// Blockchain client.
+	client: Arc<dyn Client<B>>,
+	// TODO: Still need to figure out how to pass the peerset to the handler. Can't do it in
+	// `builder.rs` as the peerset is only constructed later on in `protocol.rs`.
+	// /// Handle to use for reporting misbehaviour of peers.
+	// peerset: PeersetHandle,
+}
+
+impl<B: Block> LightClientRequestHandler<B> {
+	/// Create a new [`BlockRequestHandler`].
+	pub fn new(
+		protocol_id: &ProtocolId,
+		client: Arc<dyn Client<B>>,
+		/* peerset: PeersetHandle,*/
+	) -> (Self, ProtocolConfig) {
+		// TODO: justify 20.
+		let (tx, request_receiver) = mpsc::channel(20);
```

Comment:
Value still to be determined. For now its just copied from the block request protocol. Do we have empirical data to back-up whatever value we decide?

Reply:
I'm not sure about the answer, but imo typical usage of LC wouldn't require many simultaneous requests. I.e. making light node public and allowing anyone to call its RPC isn't a good idea - it'll just redirect all calls to the full nodes. So 20 is probably a good estimation.

Reply:
OK, I think I have confused `handler` + `sender`, sorry :) Disregard my previous comment - imo this should be similar to what full handler is using and it is using 20 either. So OK anyway :)

Reply:
Add reasoning for `20` in https://github.com/paritytech/substrate/pull/7895/commits/b39509a1688dc09b2010a2d1aa9dde2412bb4f50. Resolving this conversation for now. We might want to revisit this in the future.